hail/src/main/scala/is/hail/expr/ir/EmitFunctionBuilder.scala

```diff
@@ -171,22 +172,22 @@ class EmitFunctionBuilder[F >: Null](
     _mods += name -> mod
   }
 
-  def addHadoopConfiguration(hConf: SerializableHadoopConfiguration): Unit = {
-    assert(hConf != null)
-    if (_hconf == null) {
-      cn.interfaces.asInstanceOf[java.util.List[String]].add(typeInfo[FunctionWithHadoopConfiguration].iname)
-      val confField = newField[SerializableHadoopConfiguration]
-      val mb = new EmitMethodBuilder(this, "addHadoopConfiguration", Array(typeInfo[SerializableHadoopConfiguration]), typeInfo[Unit])
+  def addFS(fs: FS): Unit = {
+    assert(fs != null)
+    if (_hfs == null) {
+      cn.interfaces.asInstanceOf[java.util.List[String]].add(typeInfo[FunctionWithFS].iname)
+      val confField = newField[FS]
+      val mb = new EmitMethodBuilder(this, "addFS", Array(typeInfo[FS]), typeInfo[Unit])
       methods.append(mb)
-      mb.emit(confField := mb.getArg[SerializableHadoopConfiguration](1))
-      _hconf = hConf
+      mb.emit(confField := mb.getArg[FS](1))
+      _hfs = fs
       _hfield = confField
     }
-    assert(_hconf.value == hConf.value && _hfield != null)
+    assert(_hfs == fs && _hfield != null)
```

Comment:
We ( @cseed and @tpoterba , I ) spoke about this and decided reference equality was fine; we don't expect to use more than one file system object between calls to addFS / getFS